provider-ui/ui_dev_server.js

```diff
@@ -1,39 +1,49 @@
-const { createServer } = require('http');
-const { parse } = require('url');
-const next = require('next');
+const { createServer } = require('http')
+const { parse } = require('url')
+const next = require('next')
 
-const port = parseInt(process.env.PORT, 10) || 3001;
-const dev = process.env.NODE_ENV !== 'production';
-const app = next({ dev });
-const handle = app.getRequestHandler();
+const dev = process.env.NODE_ENV !== 'production'
+const hostname = 'localhost'
+const port = 3001
+// when using middleware `hostname` and `port` must be provided below
+const app = next({ dev, hostname, port })
+const handle = app.getRequestHandler()
 const httpProxy = require('http-proxy');
 
 const proxy = httpProxy.createProxyServer({});
 
 proxy.on('error', (err, req, res) => {
-  res.writeHead(500, {
-    'Content-Type': 'text/plain',
-  });
-  res.end('Proxy issue.');
+    res.writeHead(500, {
+        'Content-Type': 'text/plain',
+    });
+    res.end('Proxy issue via provider-ui.');
 });
 
-
 app.prepare().then(() => {
-  createServer((req, res) => {
-    // Be sure to pass `true` as the second argument to `url.parse`.
-    // This tells it to parse the query portion of the URL.
-    const { pathname } = parse(req.url, true);
-    if (pathname.startsWith('/api') || pathname.startsWith('/user/logout') || pathname.startsWith('/user/login')) {
-      proxy.web(req, res, { target: 'http://localhost:9081' });
-    } else {
-      if (req.url.startsWith('/provider')) {
-        req.url = req.url.replace('/provider', '');
-      }
-      handle(req, res);
-    }
-    // app.render(req, res, '/blog', Object.assign(params, query))
-  }).listen(port, (err) => {
-    if (err) throw err;
-    console.log(`> Ready on http://localhost:${port}`);
-  });
-});
+    createServer(async (req, res) => {
+        try {
+            // Be sure to pass `true` as the second argument to `url.parse`.
+            // This tells it to parse the query portion of the URL.
+            const parsedUrl = parse(req.url, true)
+            const { pathname, query } = parsedUrl
+
+            if (pathname.startsWith('/api')) {
+                proxy.web(req, res, {
+                    target: 'http://localhost:9081',
+                })
+            } else {
+                if (req.url.startsWith('/provider')) {
+                    req.url = req.url.replace('/provider', '')
+                }
+                await handle(req, res, parsedUrl)
+            }
+        } catch (err) {
+            console.error('Error occurred handling', req.url, err)
+            res.statusCode = 500
+            res.end('internal server error')
```

Comment:
Can we offer something more informative?

Reply:
Let me think of something